pkg/kubelet/metrics/metrics.go

```diff
@@ -30,46 +30,48 @@ import (
 
 // This const block defines the metric names for the kubelet metrics.
 const (
-	KubeletSubsystem                   = "kubelet"
-	NodeNameKey                        = "node_name"
-	NodeLabelKey                       = "node"
-	NodeStartupPreKubeletKey           = "node_startup_pre_kubelet_duration_seconds"
-	NodeStartupPreRegistrationKey      = "node_startup_pre_registration_duration_seconds"
-	NodeStartupRegistrationKey         = "node_startup_registration_duration_seconds"
-	NodeStartupPostRegistrationKey     = "node_startup_post_registration_duration_seconds"
-	NodeStartupKey                     = "node_startup_duration_seconds"
-	PodWorkerDurationKey               = "pod_worker_duration_seconds"
-	PodStartDurationKey                = "pod_start_duration_seconds"
-	PodStartSLIDurationKey             = "pod_start_sli_duration_seconds"
-	PodStartTotalDurationKey           = "pod_start_total_duration_seconds"
-	CgroupManagerOperationsKey         = "cgroup_manager_duration_seconds"
-	PodWorkerStartDurationKey          = "pod_worker_start_duration_seconds"
-	PodStatusSyncDurationKey           = "pod_status_sync_duration_seconds"
-	PLEGRelistDurationKey              = "pleg_relist_duration_seconds"
-	PLEGDiscardEventsKey               = "pleg_discard_events"
-	PLEGRelistIntervalKey              = "pleg_relist_interval_seconds"
-	PLEGLastSeenKey                    = "pleg_last_seen_seconds"
-	EventedPLEGConnErrKey              = "evented_pleg_connection_error_count"
-	EventedPLEGConnKey                 = "evented_pleg_connection_success_count"
-	EventedPLEGConnLatencyKey          = "evented_pleg_connection_latency_seconds"
-	EvictionsKey                       = "evictions"
-	EvictionStatsAgeKey                = "eviction_stats_age_seconds"
-	PreemptionsKey                     = "preemptions"
-	VolumeStatsCapacityBytesKey        = "volume_stats_capacity_bytes"
-	VolumeStatsAvailableBytesKey       = "volume_stats_available_bytes"
-	VolumeStatsUsedBytesKey            = "volume_stats_used_bytes"
-	VolumeStatsInodesKey               = "volume_stats_inodes"
-	VolumeStatsInodesFreeKey           = "volume_stats_inodes_free"
-	VolumeStatsInodesUsedKey           = "volume_stats_inodes_used"
-	VolumeStatsHealthStatusAbnormalKey = "volume_stats_health_status_abnormal"
-	RunningPodsKey                     = "running_pods"
-	RunningContainersKey               = "running_containers"
-	DesiredPodCountKey                 = "desired_pods"
-	ActivePodCountKey                  = "active_pods"
-	MirrorPodCountKey                  = "mirror_pods"
-	WorkingPodCountKey                 = "working_pods"
-	OrphanedRuntimePodTotalKey         = "orphaned_runtime_pods_total"
-	RestartedPodTotalKey               = "restarted_pods_total"
+	FirstNetworkPodStartSLIDurationKey   = "first_network_pod_start_sli_duration_seconds"
+	FirstNetworkPodStartTotalDurationKey = "first_network_pod_start_total_duration_seconds"
```

Comment:
The current naming is based on an implementation detail so it may be difficult to understand for the end users.
Based on my current understanding of this metric, I would suggest something along those lines:
```
kubelet_node_startup_network_initialization_delay_duration_seconds
```

Reply:
it is not really node startup since it depends on the workload to be scheduled to that node, that is the tricky part ... this metric only represent node_startup if the workload is scheduled as soon as the node is ready

Reply:
since is internal and temporary I think this name is ok since it really represents the metric